src/Illuminate/Validation/ValidationRuleParser.php

```diff
@@ -79,29 +79,39 @@ protected function explodeRules($rules)
      * Explode the explicit rule into an array if necessary.
      *
      * @param  mixed  $rule
+     * @param  string  $attribute
      * @return array
      */
-    protected function explodeExplicitRule($rule)
+    protected function explodeExplicitRule($rule, $attribute)
     {
         if (is_string($rule)) {
             return explode('|', $rule);
         } elseif (is_object($rule)) {
-            return [$this->prepareRule($rule)];
+            return [$this->prepareRule($rule, $attribute)];
         }
 
-        return array_map([$this, 'prepareRule'], $rule);
+        $attributes = array_fill(
+            array_key_first($rule), count($rule), $attribute
+        );
+
+        return array_map(
+            [$this, 'prepareRule'], $rule, $attributes
+        );
     }
 
     /**
      * Prepare the given rule for the Validator.
      *
      * @param  mixed  $rule
+     * @param  string  $attribute
      * @return mixed
      */
-    protected function prepareRule($rule)
+    protected function prepareRule($rule, $attribute)
     {
         if ($rule instanceof Closure) {
             $rule = new ClosureValidationRule($rule);
+        } elseif ($rule instanceof NestedRules) {
+            $rule = $rule->compile($attribute, $this->data[$attribute] ?? null);
```

Comment:
`$this->data[$attribute]` is it valid `$this->data` is poor array, but what if the attribute will have complex key like `data.test` ?

Reply:
Any data that is set with the attribute's key will be supplied into the `Rule::nested()` callback.

So if it's an array, it will be an array. This line you mention allows you to place a nested instance on any validation field:

```php
$rules = [
    'name' => Rule::nested(function ($attribute, $value) {
        // (string) "name"
        var_dump($attribute);
        
        // (string) "Taylor Otwell"
        var_dump($value);
    }),
];

Validator::validate(['name' => 'Taylor Otwell'], $rules);
```